source/common/tracing/http_tracer_impl.cc

```diff
@@ -135,11 +150,13 @@ HttpTracerImpl::HttpTracerImpl(DriverPtr&& driver, const LocalInfo::LocalInfo& l
 
 SpanPtr HttpTracerImpl::startSpan(const Config& config, Http::HeaderMap& request_headers,
                                   const Http::AccessLog::RequestInfo& request_info) {
-  std::string operation_name =
-      config.operationName() + " " + request_headers.Host()->value().c_str();
+  std::string span_name = HttpTracerUtility::toString(config.operationName());
+
+  if (config.operationName() == OperationName::Egress) {
+    span_name += " " + std::string(request_headers.Host()->value().c_str());
```

Comment:
Rewrite this using 2 append calls to avoid extra string constructions in the fast path.